src/document-adapter.coffee

```diff
@@ -249,54 +255,56 @@ EmberCouchDBKit.DocumentAdapter = DS.Adapter.extend
     query.options = {} unless query.options
     query.options.include_docs = true
 
-    normalizeResponce = (data) ->
+    normalizeResponse = (data) ->
       json = {}
       json[designDoc] = data.rows.getEach('doc').map((doc) => @_normalizeRevision(doc))
+      json['total_rows'] = data.total_rows
       json
 
-    @ajax('_design/%@/_view/%@'.fmt(designDoc, query.viewName), 'GET', normalizeResponce, {
+    @ajax('_design/%@/_view/%@'.fmt(designDoc, query.viewName), 'GET', normalizeResponse, {
       context: this
       data: query.options
     })
 
   findAll: (store, type) ->
 
-    typeString = Ember.String.singularize(type.typeKey)
-
+    typeString = Ember.String.singularize(type.modelName)
     designDoc = @get('designDoc') || typeString
 
     typeViewName = @get('typeViewName')
 
-    normalizeResponce = (data) ->
+    normalizeResponse = (data) ->
       json = {}
-      json[[Ember.String.pluralize(type.typeKey)]] = data.rows.getEach('doc').map((doc) => @_normalizeRevision(doc))
+      json[[Ember.String.pluralize(type.modelName)]] = data.rows.getEach('doc').map((doc) => @_normalizeRevision(doc))
       json
 
     data =
       include_docs: true
       key: '"' + typeString + '"'
 
-    @ajax('_design/%@/_view/%@'.fmt(designDoc, typeViewName), 'GET', normalizeResponce, {
+    @ajax('_design/%@/_view/%@'.fmt(designDoc, typeViewName), 'GET', normalizeResponse, {
       data: data
     })
 
-  createRecord: (store, type, record) ->
-    json = store.serializerFor(type.typeKey).serialize(record._createSnapshot())
-    @_push(store, type, record, json)
+  createRecord: (store, type, snapshot) ->
+    json = store.serializerFor(type.modelName).serialize(snapshot)
+    delete json.rev
```

Comment:
We had a need for strict schema in our files, and the original added a new "rev" attribute to the document.